File: pages/home_page.py
class HomePage:

    def __init__(self, page):
        self.page = page

        # Locators
        self.about_link = page.get_by_role("link", name="אודות")
        self.contact_link = page.get_by_role("link", name="צור קשר")

    def go_to_about(self):
        self.about_link.click()

    def go_to_contact(self):
        self.contact_link.click()

    def go_to_login(self):
        self.page.get_by_role("button", name="התחבר/י").click()

File: pages/home-page.ts
import { Page } from "@playwright/test";
import { closePopupIfExists } from "../utils/popup-handler";
import { BasePage } from "./base-page";

export class HomePage extends BasePage {
  private readonly aboutLink = this.page.getByRole("link", { name: "אודות" });
  private readonly contactLink = this.page.getByRole("link", { name: "צור קשר" });
  private readonly loginButton = this.page.getByRole("button", { name: "התחבר/י" });
  private readonly profileLink = this.page.getByRole("link", { name: /הי/ });

  async navigate(): Promise<void> {
    await super.navigate("/");
    await closePopupIfExists(this.page);
  }

  async goToAbout(): Promise<void> {
    await this.aboutLink.click();
  }

  async goToContact(): Promise<void> {
    await this.contactLink.click();
  }

  async goToLogin(): Promise<void> {
    await this.loginButton.click();
  }

  async goToCart(): Promise<void> {
    await this.page.getByRole("link", { name: "עגלת קניות" }).click();
  }

  async goToProfile(): Promise<void> {
    await this.profileLink.click();
  }
}
